sdk/eventhub/event-hubs/README.md

```diff
@@ -181,6 +182,46 @@ for await (const events of consumer.getEventIterator()){
 }
 ```
 
+### Consume events using an Event Processor
+
+To consume events for all partitions of an Event Hub, create an `EventProcessor` for a specific consumer group. When an Event Hub is created, it provides a default consumer group that can be used to get started.
+
+The `EventProcessor` will delegate the processing of events to a `PartitionProcessor` that you provide, allowing you to focus on business logic while the processor holds responsibility for managing the underlying consumer operations. In our example, we will focus on building the `EventProcessor` and use a very minimal partition processor that does no actual processing.
+
+```javascript
+class SimplePartitionProcessor {
+  async processEvents(events: ReceivedEventData[]) {
+    // your code here
+  }
+
+  async processError(error: Error) {
+    // your error handler here
+  }
+}
+
+const processor = new EventProcessor(
+  EventHubClient.defaultConsumerGroupName,
+  client,
+  () => {
+    return new SimplePartitionProcessor();
+  },
+  new InMemoryPartitionManager()
+);
+await processor.start();
+// At this point, the processor is consuming events from each partition of the Event Hub and
+// delegating them to the SimplePartitionProcessor instance created for that partition.  This
+// processing takes place in the background and will not block.
+//
+// It is important to note that the processor does not own the EventHubClient that was passed into it.
+// You are responsible for ensuring that it is disposed after processing.  It is also important that it
+// not be closed or disposed during the time that the EventProcessor is running.
```

Comment:
Can we pull this note on EventHubClient out of the code sample? It doesnt feel like the right place for it